modules/auxiliary/gather/ldap_esc1_cert_finder.rb

```diff
@@ -0,0 +1,420 @@
+        attributes_object_present = ['displayName', 'name']
+        entries_object_present = query_ldap_server(raw_filter_object_present, attributes_object_present, base: nil, base_prefix: 'CN=Configuration')
+
+        print_status("   ObjectType: #{object_type}")
+        print_status("      DN: #{entries_object_present[0][:dn][0]}")
+        print_status("      Display Name: #{entries_object_present[0][:displayname][0]}") if entries_object_present[0][:displayname]
+        print_status("      Name: #{entries_object_present[0][:name][0]}") if entries_object_present[0][:name]
+      end
+
+      if ace_body[:flags] && ace_body[:flags][:ace_inherited_object_type_present] == 1
+        print_status("   InheritedObjectType: #{ace_body[:inherited_object_type]}")
+      end
+    end
+
+    [flag_allowed_to_enroll, allowed_sids]
+  end
+
+  def query_ldap_server(raw_filter, attributes, base: nil, base_prefix: nil)
+    ldap_connect do |ldap|
+      bind_result = ldap.as_json['result']['ldap_result']
+
+      # Codes taken from https://ldap.com/ldap-result-code-reference-core-ldapv3-result-codes
+      case bind_result['resultCode']
+      when 0
+        print_good('Successfully bound to the LDAP server!')
+      when 1
+        fail_with(Failure::NoAccess, "An operational error occurred, perhaps due to lack of authorization. The error was: #{bind_result['errorMessage']}")
+      when 7
+        fail_with(Failure::NoTarget, 'Target does not support the simple authentication mechanism!')
+      when 8
+        fail_with(Failure::NoTarget, "Server requires a stronger form of authentication than we can provide! The error was: #{bind_result['errorMessage']}")
+      when 14
+        fail_with(Failure::NoTarget, "Server requires additional information to complete the bind. Error was: #{bind_result['errorMessage']}")
+      when 48
+        fail_with(Failure::NoAccess, "Target doesn't support the requested authentication type we sent. Try binding to the same user without a password, or providing credentials if you were doing anonymous authentication.")
+      when 49
+        fail_with(Failure::NoAccess, 'Invalid credentials provided!')
+      else
+        fail_with(Failure::Unknown, "Unknown error occurred whilst binding: #{bind_result['errorMessage']}")
+      end
+      if @base_dn
+        print_good("Using already dicovered base DN: #{@base_dn}")
+      elsif (@base_dn = base)
+        print_status("Using user-specified base DN: #{@base_dn}")
+      else
+        print_status('Discovering base DN automatically')
+
+        unless (@base_dn = discover_base_dn(ldap))
+          print_warning("Couldn't discover base DN!")
+        end
+      end
+
+      begin
+        filter = Net::LDAP::Filter.construct(raw_filter)
+      rescue StandardError => e
+        fail_with(Failure::BadConfig, "Could not compile the filter to find the ESC1 objects! Error was #{e}")
+      end
+
+      base ||= @base_dn
+      full_base_dn = "#{base_prefix},#{base}"
```

Comment:
Good point this will result in errors. I think the only reason I didn't notice this before was that the queries that I was using all had a `base_prefix` specified already. However you are right this would mess up the logic here.

Reply:
Added some more logic in here to more appropriately validate these two parameters and react accordingly.

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/17122/commits/051b5f6494aea85ea9c1944dbf6c1eb238185ac0